src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -156,6 +160,12 @@ private static JsonConverter[] GetDefaultFactoryConverters()
         /// </remarks>
         public IList<JsonPolymorphicTypeConfiguration> PolymorphicTypeConfigurations => _polymorphicTypeConfigurations;
 
+        internal static void IntializeConverterProviderForSourceGen()
+        {
+            s_converterRetrievalFunc ??= (type, options) => options.GetConverterForTypeSourceGen(type);
```

Comment:
Perhaps I'm missing something but assuming GetConverterForTypeSourceGen is linker safe, why can't we just call it directly? Shouldn't reflection path be hidden under the callback. Isn't adding annotations as not going through delegate here?